src/mainwindow.cpp

```diff
@@ -200,6 +200,22 @@ MainWindow::MainWindow(QWidget* parent)
         connect(Settings::instance(), &Settings::prettifySymbolsChanged, this, [this](bool prettifySymbols) {
             m_config->group("Settings").writeEntry("prettifySymbols", prettifySymbols);
         });
+
+        settings->setPaths(
+            m_config->group("Flamegraph").readEntry("userPaths", QDir::homePath().split(QLatin1Char(';'))),
+            config.readEntry("systemPaths", QDir::rootPath().split(QLatin1Char(';'))));
+        connect(Settings::instance(), &Settings::pathsChanged, this, [this, settings] {
+            m_config->group("Flamegraph").writeEntry("userPaths", settings->userPaths().join(QLatin1Char(';')));
```

Comment:
don't join manually, just write the string list directly

https://api.kde.org/frameworks-api/frameworks-apidocs/frameworks/kconfig/html/classKConfigGroup.html#a2b935a52d80b2e1af6aa3f26ee0833ef